lib/postgrex/protocol.ex

```diff
@@ -3098,7 +3116,14 @@ defmodule Postgrex.Protocol do
         msg_recv(s, timeout, IO.iodata_to_binary([buffer | data]))
 
       {:error, reason} ->
-        disconnect(s, tag(mod), "recv", reason, IO.iodata_to_binary(buffer))
+        action =
+          if s.postgres == :idle do
+            "recv (idle)"
```

Comment:
this is so that we have a better idea where the error came from:

<img width="1037" alt="image" src="https://user-images.githubusercontent.com/76071/153901999-c5acbda5-9ad6-4a0a-9927-2c252bcdb26b.png">